### Ques 1

The key difference between a t-test and a z-test lies in the type of data and the sample size. Both tests are used for hypothesis testing about population means. However, the choice between t-test and z-test depends on whether the population standard deviation is known and the sample size.

T-test: Used when the population standard deviation is unknown or when the sample size is small. There are two main types: one-sample t-test, independent two-sample t-test, and paired (dependent) two-sample t-test. Example: Testing whether a new drug has an effect on blood pressure in a small sample of patients.

Z-test: Used when the population standard deviation is known and the sample size is large (typically above 30). Example: Testing whether the average height of a population is significantly different from a known standard height when you have a large dataset.

### Ques 2

- One-tailed test: It tests the hypothesis in one direction only (greater than or less than). Example: Testing whether a new teaching method improves test scores compared to the old method.
- Two-tailed test: It tests the hypothesis in both directions (greater than and less than). Example: Testing whether a new drug has any effect on a disease.

### Ques 3

- Type 1 error (False Positive): Occurs when you reject a true null hypothesis. This is also called a "false alarm." 
- Type 2 error (False Negative): Occurs when you fail to reject a false null hypothesis. This is also called a "miss." 

An example of a Type 1 error is to conclude that there is a difference in the average test scores of group A and group B when there is actually no difference. An example of a Type 2 error is to conclude that there is no difference in the average test scores of group A and group B when there actually is a difference.

### Ques 4

Bayes's theorem is a formula that can be used to update the probability of an event based on new information.

For example, let's say you know that 10% of the population has a certain disease. You also know that a test for the disease has a 90% accuracy rate. This means that if you have the disease, the test will be positive 90% of the time. And if you don't have the disease, the test will be negative 90% of the time.

Now, let's say you take the test and it is positive. What is the probability that you actually have the disease?

Using Bayes's theorem, we can calculate that the probability of having the disease given a positive test result is 56.25%. This is because the probability of a positive test result is 90%, and the probability of having the disease given a positive test result is 10%.

### Ques 5

A confidence interval is a range of values that is likely to contain the true population mean. The confidence interval is calculated by taking the sample mean and adding or subtracting a margin of error.

The margin of error is calculated using the following formula:

margin of error = z * standard error of the mean
where:

z is the z-score for the desired confidence level
standard error of the mean is the standard deviation of the sample divided by the square root of the sample size
For example, let's say we want to calculate a 95% confidence interval for the average height of adult males. The z-score for a 95% confidence level is 1.96. The standard error of the mean is 2 inches.

Plugging these values into the formula, we get a margin of error of 1.96 inches.

The confidence interval is then calculated by taking the sample mean and adding and subtracting the margin of error. In this case, the confidence interval is 67.04 to 68.96 inches.

This means that we are 95% confident that the true population mean height of adult males is within the range of 67.04 to 68.96 inches.

### Ques 6

Bayes' Theorem is a fundamental concept in probability theory and statistics used to update our beliefs about the probability of an event occurring when we have new evidence. 

Problem:

A doctor is testing a new blood test for cancer. The blood test is 90% accurate in detecting cancer when a person actually has cancer. However, the blood test is also 10% accurate in detecting cancer when a person does not actually have cancer.

The doctor knows that 1% of the population has cancer. If the blood test for a patient is positive, what is the probability that the patient actually has cancer?

Solution:

We can use Bayes' Theorem to calculate the probability that the patient actually has cancer given that the blood test is positive. Bayes' Theorem states that the probability of event A occurring given event B is equal to the probability of event A and event B occurring divided by the probability of event B occurring.

In this case, event A is the patient having cancer and event B is the blood test being positive.

The probability of event A and event B occurring is equal to the probability of the blood test being positive given that the patient has cancer times the probability of the patient having cancer.

The probability of the blood test being positive given that the patient has cancer is 90%.

The probability of the patient having cancer is 1%.
The probability of event B occurring is equal to the probability of the blood test being positive given that the patient does not have cancer plus the probability of the blood test being positive given that the patient does have cancer.

The probability of the blood test being positive given that the patient does not have cancer is 10%.

The probability of the blood test being positive given that the patient does have cancer is 90%.

Therefore, the probability of the patient actually having cancer given that the blood test is positive is equal to (90% * 1%) / (90% * 1% + 10% * 99%) = 9%.

In other words, there is a 9% chance that the patient actually has cancer if the blood test is positive.

### Ques 7

In [1]:
import numpy as np
import scipy.stats as stats

mean=50
std=5
confidence_level=0.95

z_score = stats.norm.ppf(1 - (1 - confidence_level) / 2)

confidance_interval=(mean-z_score*std,mean+z_score*std)

print(confidance_interval)

(40.200180077299734, 59.799819922700266)


### Ques 8


The margin of error in a confidence interval is the amount of uncertainty that is associated with the estimate. It is calculated by taking the standard deviation of the sample and multiplying it by the z-score for the desired confidence level.

The sample size affects the margin of error in a way that is inversely proportional. This means that as the sample size increases, the margin of error decreases.

In [2]:
import numpy as np
import scipy.stats as st

mean=50
std=10
CI=0.95

MOE_100=st.norm.ppf(1 - (1 -CI) / 2)*std/np.sqrt(100)
MOE_300=st.norm.ppf(1-(1-CI)/2)*std/np.sqrt(300)

print("Margin of error for sample size of 100:", MOE_100)
print("Margin of error for sample size of 300:", MOE_300)


Margin of error for sample size of 100: 1.9599639845400538
Margin of error for sample size of 300: 1.1315857340761715


### Ques 9

In [3]:
import numpy as np
import scipy.stats as st

data_point=75
p_mean=70
p_std=5
Z_scr=(data_point-p_mean)/p_std
print('Z_score for data point with a value of 75 is',Z_scr)

Z_score for data point with a value of 75 is 1.0


Interpretation:
The calculated z-score is 1. This means that the data point with a value of 75 is 1 standard deviation above the population mean of 70. In other words, it is one "standard deviation unit" away from the mean. This information helps us to understand the relative position of the data point within the population's distribution.

A positive z-score indicates that the data point is above the mean, while a negative z-score would indicate that the data point is below the mean. The magnitude of the z-score tells us how many standard deviations the data point is away from the mean. In this case, a z-score of 1 means the data point is 1 standard deviation above the mean.

### Ques 10

Hypothesis test for the effectiveness of the new weight loss drug:

1. Hypothesis:
The null hypothesis is that the drug is not effective. mean =0
The alternative hypothesis is that the drug is effective. mean !=0

2. Significance level:

The significance level is 0.05, which means that we are 95% confident in our results.

3. Test statistic:

The t-statistic is calculated as the difference between the sample mean and the hypothesized population mean, divided by the standard deviation of the sample.

t = (sample mean - hypothesized population mean) / (standard deviation of the sample)

In [4]:
import numpy as np
import scipy.stats as st

sample_size=50
sample_mean=6
sample_std=2.5
CI=.95
hypothesized_mean=0

t=(sample_mean-hypothesized_mean)/(sample_std/np.sqrt(sample_size))
print('t statistic:',t)

t statistic: 16.970562748477143


4. Critical value:

The critical value is the value of the t-statistic that we need to exceed in order to reject the null hypothesis. The critical value is calculated using the significance level and the degrees of freedom.

In [5]:
critical_value=st.t.ppf((1-(1-CI)/2),df=sample_size-1)

In [6]:
critical_value

2.009575234489209

5. Decision rule:

We reject the null hypothesis if the t-statistic is greater than the critical value. In this case, the t-statistic (16.97) is greater than the critical value (2.009), so we reject the null hypothesis.

In [7]:
# Make the decision
if t > critical_value:
  print("Reject the null hypothesis")
else:
  print("Fail to reject the null hypothesis")

Reject the null hypothesis


6. Conclusion:

We can conclude that the drug is significantly effective at a 95% confidence level.

### Ques 11

To calculate the 95% confidence interval for the true proportion of people who are satisfied with their job:

1. Calculate the critical value \(Z\) for a 95% confidence interval (standard normal distribution), which is approximately 1.96.

2. Calculate the standard error (SE):
   - SE = \(\sqrt{\frac{0.65 \times (1 - 0.65)}{500}}\)

3. Calculate the margin of error (ME):
   - ME = \(Z \times SE\)

4. Calculate the lower and upper bounds of the confidence interval:
   - Lower Bound = \(\hat{p} - ME\)
   - Upper Bound = \(\hat{p} + ME\)

Let's perform the calculations:

- \(Z = 1.96\) (for a 95% confidence interval).
- \(SE = \sqrt{\frac{0.65 \times (1 - 0.65)}{500}} \approx 0.021\).
- \(ME = 1.96 \times 0.021 \approx 0.0418\).

Now, calculate the confidence interval:

- Lower Bound = \(0.65 - 0.0418 \approx 0.6082\) (or 60.82%)
- Upper Bound = \(0.65 + 0.0418 \approx 0.6918\) (or 69.18%)

So, the 95% confidence interval for the true proportion of people who are satisfied with their job is approximately 60.82 to 69.18%.


In [8]:
import scipy.stats as stats
import numpy as np

sample_proportion=0.65
sample_size=500

CI=0.95

z_critical=stats.norm.ppf(1-(1-CI)/2)

standard_error=np.sqrt((sample_proportion*(1-sample_proportion))/sample_size)

moe=z_critical*standard_error

lower_bound=sample_proportion-moe
upper_bound=sample_proportion+moe

print(f"Lower Bound: {lower_bound:.4f} (or {lower_bound * 100:.2f}%)")
print(f"Upper Bound: {upper_bound:.4f} (or {upper_bound * 100:.2f}%)")

Lower Bound: 0.6082 (or 60.82%)
Upper Bound: 0.6918 (or 69.18%)


### Ques 12

#### Step 1: Define the Hypotheses

- **Null Hypothesis (H0):** The two teaching methods have no significant difference in student performance. (μ1 - μ2 = 0)
- **Alternative Hypothesis (Ha):** The two teaching methods do have a significant difference in student performance. (μ1 - μ2 ≠ 0)


In [9]:
# Define hypotheses
null_hypothesis = "The two teaching methods have no significant difference in student performance."
alternative_hypothesis = "The two teaching methods do have a significant difference in student performance."

#### Step 2: Set the Significance Level

- Significance level (α) = 0.01


In [10]:
alpha = 0.01

#### Step 3: Collect and Prepare Data

- Sample A: Mean score (x̄1) = 85, Standard Deviation (σ1) = 6
- Sample B: Mean score (x̄2) = 82, Standard Deviation (σ2) = 5
- sample_size=30 (Assume)

In [11]:
# sample data
mean_A=85
std_A=6
mean_B=82
std_B=5
n1=n2=30

#### Step 4: Calculate the Test Statistic
- The formula for the two-sample t-test is given as:
- t = (x̄1 - x̄2) / sqrt((s1^2/n1) + (s2^2/n2))
- Where s1 and s2 are the sample standard deviations, n1 and n2 are the sample sizes, x̄1 and x̄2 are the sample means.

In [12]:
t_stats=(mean_A-mean_B)/np.sqrt((std_A**2/n1)+(std_B**2/n2))
t_stats

2.1038606199548298

#### Step 5: Determine the Critical Value

- Degrees of freedom (df) = n1 + n2 - 2
- Using a t-table or a statistical calculator, find the critical t-value for a two-tailed test with α/2 = 0.005 and df degrees of freedom.


In [13]:
dof=n1+n2-2
critical_t=stats.t.ppf(1-alpha/2,dof)
critical_t

2.6632869538098674

#### Step 6: Compare the Test Statistic and Critical Value
- Compare the test statistic and critical value

In [14]:
if abs(t_stats)>critical_t:
    decision="Reject null hypothesis"
else:
    decision="Fail to reject null hypothesis"
decision

'Fail to reject null hypothesis'

#### Step 7: Calculate the p-value

- Using a t-table or a statistical calculator, find the p-value associated with the calculated t statistic.


In [15]:
p_value = 2 * (1 - stats.t.cdf(abs(t_stats), dof))

# Display the decision and p-value
print(f"Decision: {decision}")
print(f"P-Value: {p_value:.4f}")

Decision: Fail to reject null hypothesis
P-Value: 0.0397


#### Step 8: Make a Decision
- Determine whether to reject the null hypothesis based on the p-value

In [16]:
if p_value <alpha:
    decision = "Reject null hypothesis"
else:
    decision = "Fail to reject null hypothesis"

#### Step 9: Interpret the Results
- Provide an interpretation based on the decision


In [17]:
if decision == "Reject null hypothesis":
    interpretation = "There is a significant difference in student performance between the two teaching methods."
else:
    interpretation = "There is no significant difference in student performance between the two teaching methods."

# Display the final decision and interpretation
print(f"Decision: {decision}")
print(f"Interpretation: {interpretation}")

Decision: Fail to reject null hypothesis
Interpretation: There is no significant difference in student performance between the two teaching methods.


### Ques 13

In [18]:
import scipy.stats as stats
import math

# Sample data
sample_mean = 65
sample_std_dev = 8
sample_size = 50
confidence_level = 0.90

# Step 1: Calculate the critical value (Z) for a 90% confidence interval
alpha = 1 - confidence_level
critical_value = stats.norm.ppf(1 - alpha / 2)  # Two-tailed test

# Step 2: Calculate the confidence interval
margin_of_error = critical_value * (sample_std_dev / math.sqrt(sample_size))
lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

# Display the confidence interval
print(f"Confidence Interval: ({lower_bound:.2f}, {upper_bound:.2f})")


Confidence Interval: (63.14, 66.86)


### Step 1: Define the Hypotheses

- **Null Hypothesis (H0):** Caffeine has no significant effect on reaction time. (μ = μ0, where μ0 is the hypothesized population mean)
- **Alternative Hypothesis (Ha):** Caffeine has a significant effect on reaction time. (μ ≠ μ0)


### Step 2: Set the Significance Level

- Significance level (α) = 0.10 (for a 90% confidence level)


In [19]:
significance_level = 0.10

### Step 3: Collect and Prepare Data

- Sample mean (x̄) = 0.25 seconds
- Sample standard deviation (s) = 0.05 seconds
- Sample size (n) = 30


In [20]:
sample_mean = 0.25
sample_std_dev = 0.05
sample_size = 30

### Step 4: Calculate the Test Statistic

- The formula for the one-sample t-test is given as:
  - t = \(\frac{x̄ - μ0}{\frac{s}{\sqrt{n}}}\)
  - Where \(x̄\) is the sample mean, μ0 is the hypothesized population mean, s is the sample standard deviation, and n is the sample size.


In [21]:
hypothesized_mean = 0.26  
t_statistic = (sample_mean - hypothesized_mean) / (sample_std_dev / (sample_size**0.5))

print(f"Test Statistic: {t_statistic:.4f}")


Test Statistic: -1.0954


### Step 5: Determine the Critical Value

- Degrees of freedom (df) = n - 1 (for a one-sample t-test)
- Using a t-table or a statistical calculator, find the critical t-value for a two-tailed test with \(\frac{α}{2}\) = 0.05 and df degrees of freedom.


In [22]:
degrees_of_freedom = sample_size - 1

alpha_over_2 = significance_level / 2
critical_t_value = stats.t.ppf(1 - alpha_over_2, degrees_of_freedom)

print(f"Critical t-Value: {critical_t_value:.4f}")


Critical t-Value: 1.6991


### Step 6: Compare the Test Statistic and Critical Value

- If |t| > critical t-value, reject the null hypothesis (H0).
- If |t| ≤ critical t-value, fail to reject the null hypothesis (H0).

### Step 7: Calculate the p-value

- Using a t-table or a statistical calculator, find the p-value associated with the calculated t statistic.


In [23]:
if abs(t_statistic) > critical_t_value:
    decision = "Reject null hypothesis"
else:
    decision = "Fail to reject null hypothesis"

p_value = 2 * (1 - stats.t.cdf(abs(t_statistic), degrees_of_freedom))

print(f"Decision: {decision}")
print(f"P-Value: {p_value:.4f}")

Decision: Fail to reject null hypothesis
P-Value: 0.2823


### Step 8: Make a Decision

- If p-value < α, reject the null hypothesis (H0).
- If p-value ≥ α, fail to reject the null hypothesis (H0).

### Step 9: Interpret the Results

- Provide an interpretation based on the decision.


In [24]:

if p_value < significance_level:
    decision = "Reject null hypothesis"
else:
    decision = "Fail to reject null hypothesis"

if decision == "Reject null hypothesis":
    interpretation = "Caffeine has a significant effect on reaction time."
else:
    interpretation = "There is no significant effect of caffeine on reaction time."

print(f"Decision: {decision}")
print(f"Interpretation: {interpretation}")


Decision: Fail to reject null hypothesis
Interpretation: There is no significant effect of caffeine on reaction time.
